sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -1351,6 +1351,28 @@ object RepartitionByExpression {
   }
 }
 
+/**
+ * This operator does not guarantee the output partitioning, because the partition number will be
```

Comment:
let's explain what it does. We can copy something from https://github.com/apache/spark/pull/32932/files#diff-4df7a54d6e4cfdbb5b68f2d2bcb083c2c692514b53c8fd7d61c24363061bb51cR56